src/material/table/table-data-source.ts

```diff
@@ -148,6 +148,14 @@ export class MatTableDataSource<T> extends DataSource<T> {
       let valueA = this.sortingDataAccessor(a, active);
       let valueB = this.sortingDataAccessor(b, active);
 
+      const valueAType = typeof valueA;
+      const valueBType = typeof valueB;
+      const sameType = valueAType === valueBType;
+      if (!sameType) {
```

Comment:
Since `sameType` is only used once and is fairly straightforward to follow, you can inline it here.